diesel/src/upsert/on_conflict_extension.rs

```diff
@@ -76,8 +79,12 @@ where
     /// ```
     pub fn on_conflict_do_nothing(
         self,
-    ) -> InsertStatement<T, OnConflictValues<U::ValueClause, NoConflictTarget, DoNothing>, Op, Ret>
-    {
+    ) -> InsertStatement<
+        T,
+        OnConflictValues<U::ValueClause, NoConflictTarget, DoNothing, NoWhereClause>,
```

Comment:
With the added default value for the new generic argument this change is not necessary anymore.